src/overlays/actors/ovl_En_Encount3/z_en_encount3.c

```diff
@@ -32,22 +34,173 @@ ActorInit En_Encount3_InitVars = {
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Encount3/func_809AD1EC.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Encount3/EnEncount3_Update.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Encount3/EnEncount3_Draw.s")
+s32 D_809AD810 = false;
+
+void EnEncount3_Init(Actor* thisx, PlayState* play) {
+    EnEncount3* this = THIS;
+
+    this->unk14A = ENCOUNT3_GET_SPAWN_INDEX(thisx);
+    this->unk158 = ENCOUNT3_GET_PARAM_F80(thisx);
+    this->unk15C = ENCOUNT3_GET_PARAM_7F(thisx);
+    this->unk15A = this->actor.csId;
+    this->unk16C = this->actor.world.rot.z * 10.0f;
+    if (this->unk16C < 10.0f) {
+        this->unk16C = 10.0f;
+    } else if (this->unk16C > 1000.0f) {
+        this->unk16C = 1000.0f;
+    }
+    if (this->unk15C == 0x7F) {
+        this->unk15C = -1;
+    }
+    if ((this->unk15C >= 0) && (Flags_GetSwitch(play, this->unk15C) != 0)) {
+        Actor_Kill(&this->actor);
+    }
+    this->actor.flags |= ACTOR_FLAG_CANT_LOCK_ON;
+    this->actor.flags &= ~ACTOR_FLAG_TARGETABLE;
+    func_809AD058(this);
+}
+
+void EnEncount3_Destroy(Actor* thisx, PlayState* play) {
+}
+
+void func_809AD058(EnEncount3* this) {
+    this->unk154 = 0x113;
+    this->unk150 = 1;
+    this->unk15E = 0x1E;
+    this->actionFunc = func_809AD084;
+}
+
+void func_809AD084(EnEncount3* this, PlayState* play) {
+    if ((this->unk15C >= 0) && (Flags_GetSwitch(play, this->unk15C) != 0)) {
+        Actor_Kill(&this->actor);
+    } else if (!(this->unk16C < this->actor.xzDistToPlayer) && (Player_GetMask(play) == 6) && !D_809AD810) {
+        if (this->unk15E > 0) {
+            this->unk15E--;
+        } else {
+            this->unk17C =
+                Actor_SpawnAsChild(&play->actorCtx, &this->actor, play, this->unk154, this->actor.world.pos.x,
+                                   this->actor.world.pos.y, this->actor.world.pos.z, 0, 0, 0, this->unk158);
+            if (this->unk17C != NULL) {
+                this->unk14E++;
+                D_809AD810 = true;
+                this->actionFunc = func_809AD194;
+            }
+        }
+    }
+}
+
+void func_809AD194(EnEncount3* this, PlayState* play) {
+    if (this->unk14E == 0) {
+        this->unk178 = 0.0f;
+        if (this->unk15C >= 0) {
+            Flags_SetSwitch(play, this->unk15C);
+        }
+        this->actionFunc = func_809AD1EC;
+    }
+}
+
+void func_809AD1EC(EnEncount3* this, PlayState* play) {
+    if (this->unk174 < 0.002f) {
+        D_809AD810 = false;
+        Actor_Kill(&this->actor);
+    }
+}
+
+void EnEncount3_Update(Actor* thisx, PlayState* play2) {
+    EnEncount3* this = THIS;
+    f32 new_var;
+    PlayState* play = play2;
+    Player* player = GET_PLAYER(play);
+    f32 sp3C;
+    f32 sp38;
+
+    this->actionFunc(this, play);
+    if (this->actionFunc == func_809AD194) {
+        new_var = this->unk16C * 0.52f;
+        if (new_var < this->actor.xzDistToPlayer) {
```

Comment:
That local is suspicious, did you try removing it to see if it was actually necessary?

(If it is necessary that is fine.)

Reply:
without it's not working